### PR TITLE
Speed up heal-eth-prices by making requests concurrent

### PR DESCRIPTION
**Pull request by ckoopmann**
The `heal-eth-prices` command (`cargo run --bin heal-eth-prices`) was pretty slow on my machine, since it was requesting and storing every single price sequentially from the BitBy api. 

This pr ensures concurrent requests (up to a limit designed to avoid `tcp too many open connections` errors and the like), leading to a significant speed up (over 10x perceived on my machine).

Note that if any of the requests fail (for example due to rate limiting etc. ) the script will continue to run, which can lead to gaps in the price history.

### CODE REVIEW

**Comment by alextes** [comment]
@ckoopmann I feel quite hesitant to have the max concurrency, resulting in failing requests, resulting in price gaps. I'd like this to be very reliable.

How about a different approach. ~8 requests in parallel, respecting 429s (or whatever other response we get when we overload bybit) with retries that use exponential backoff?

**Comment by ckoopmann** [comment]
> @ckoopmann I feel quite hesitant to have the max concurrency, resulting in failing requests, resulting in price gaps. I'd like this to be very reliable.
> 
> How about a different approach. ~8 requests in parallel, respecting 429s (or whatever other response we get when we overload bybit) with retries that use exponential backoff?

As suggested I limited to 8 parallel request and used the [backoff](https://crates.io/crates/backoff) crate to implement an exponential backoff.

Also changed it such that the program will panic if any price cannot be loaded.

File: src/usd_price/bybit.rs [hunk 1]
```diff
@@ -52,9 +54,22 @@ async fn get_eth_candles(
...
         .format_url();
 
+    let op = || async {
+        let result = send_eth_price_request(url.clone())
+            .await
+            .map_err(Error::transient)?;
+        Ok(result)
+    };
+    let backoff = ExponentialBackoffBuilder::new()
+        .with_max_elapsed_time(Some(std::time::Duration::from_secs(60)))
```
Comment (ckoopmann) [inline src/usd_price/bybit.rs:64]: I only adjusted the `max_elapsed_time` parameter which is 15min by default.
You can find all of the default parameter values [here](https://docs.rs/backoff/latest/backoff/default/index.html).

File: src/usd_price/heal.rs [hunk 2]
```diff
@@ -57,9 +60,15 @@ pub async fn heal_eth_prices() {
...
 
-    for minute_n in 0..minutes_since_london {
-        let timestamp = london_minute_timestamp + minute_n * 60;
-        if !known_minutes.contains(&timestamp) {
+    let missing_minutes_timestamps = (0..minutes_since_london)
+        .map(|minutes| london_minute_timestamp + minutes * 60)
+        .filter(|timestamp| !known_minutes.contains(timestamp))
+        .collect::<Vec<i64>>();
+
+    let concurrent_requests = 8;
```
Comment (alextes) [inline src/usd_price/heal.rs:68]: For the future. This is statically known. In general I'd suggest having it at the top level of the module, const, and UPPERCASE.

Comment (alextes) [inline src/usd_price/heal.rs:68]: For the future. This is statically known. In general I'd suggest having it at the top level of the module, const, and UPPERCASE.

File: src/usd_price/bybit.rs (hunk 1 above)
Comment (ckoopmann) [reply src/usd_price/bybit.rs]: As discussed change it back to the default of 15mins:
https://github.com/ultrasoundmoney/eth-analysis-rs/pull/9/commits/ce2aa21fbf8338915ddaa3e8a890abb568491c63

File: src/usd_price/heal.rs (hunk 2 above)
Comment (ckoopmann) [reply src/usd_price/heal.rs]: Done:
https://github.com/ultrasoundmoney/eth-analysis-rs/pull/9/commits/ce2aa21fbf8338915ddaa3e8a890abb568491c63

**Pull request merged**